### PR TITLE
Add ospf parsing excercises

### PR DESCRIPTION
**Pull request by oianson**

### CODE REVIEW

File: 02-parsing/ospf2.robot [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Suite Setup     suite-setup
+
+*** Test Cases ***
+
+Get the ospf neighbour ID from R1 using TextFSM keywords
+    select device "r1"
+    # runs a command and parses it through the TextFSM template provided
+    # cxta includes a large collection of templates (view the 'Command Map' page in the documentation
+    # though you are able to specify your own template, as we do in this example
+    run parsed "show ip ospf neighbor" with template "02-parsing/show_ip_ospf_neighbor.textfsm"
```
Comment (oianson) [inline 02-parsing/ospf2.robot]: Not sure if @oboehmer  would like a different path here or using currentdir or anything.

File: 02-parsing/ospf1.robot [hunk 2]
```diff
@@ -7,23 +8,56 @@ Resource  cxta.robot
...
+    log list  ${output}
+    # uses another builtin keyword to set the variable called 'NBR_ID' as the first item in the list.
+    # in this example, the first item is the neighbor ID (lists starts from index 0)
+    ${NBR_ID}=  Set Variable  ${output}[0]
+    # logs the following message along with the neighbor id onto the console
+    log to console  The Neighbor ID is ${NBR_ID}
+
+Get the ospf neighbor ID state from R1
+    # uses a longer regex pattern to retrieve the ospf state
+    # we don't need to use the 'run' keyword again to execute the command as it was
```
Comment (oboehmer) [inline 02-parsing/ospf1.robot]: In the first exercise (01-basic), we highlight that test cases should be indepenent from each other. So can you please re-execute the run command, maybe adding a comment that it it not needed if the whole test suite is executed, but done as best practice..

File: 02-parsing/ospf1.robot [hunk 3]
```diff
@@ -7,23 +8,56 @@ Resource  cxta.robot
...
+    # the last output executed and therefore still in the buffer
+    ${output}=  extract pattern "\\d+\.\\d+\.\\d+\.\\d+\\s+\\d+\\s+([\\w\/]+)"
+    log to console  The OSPF neighbor has the state of "${output}"
+    Should be equal  ${output}   FULL/BDR
+
+Check the ospf neighbor ID state is FULL/BDR
+    # instead of extracting patterns, we can verify that the output matches a pattern
```
Comment (oboehmer) [inline 02-parsing/ospf1.robot:54]: same comment as above

File: 02-parsing/ospf2.robot [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    connect to all devices
+
+Check if OSPF hello interval on device "${device}" interface "${interface}" is "${interval}"
+    # uses a pyats keyword to learn operational information on ospf
+    # the keyword runs various ospf commands as per the platforms model in genie
+    ${d}=  learn "ospf" on device "r1"
+    ${v}=  Get Value From Json   ${d.to_dict()}   $..interfaces.${interface}.hello_interval
+    Should be equal as numbers   ${v[0]}  ${interval}
+
+*** Variables ***
```
Comment (oboehmer) [inline 02-parsing/ospf2.robot]: would you mind moving the variables section up, right after settings? IMHO easier to read/understand this way.

**Review by oboehmer — CHANGES_REQUESTED** [review]
thanks again for the contribution, Oli! A few minor comments..

File: 02-parsing/ospf2.robot (hunk 4 above)
Comment (oianson) [reply 02-parsing/ospf2.robot]: That's fine ... keywords above test cases IMO also is best :) 

will leave them at the bottom of consistency though with 01-test2.robot  unless you agree and we could change.

File: 02-parsing/ospf1.robot (hunk 2 above)
Comment (oianson) [reply 02-parsing/ospf1.robot]: Done, also re-executed 'select device'

**Review by oboehmer — APPROVED** [review]

**Pull request merged**